Feat/#6 create the repo interface and the service class

sebastianbadea197 opened the pull request:
Add the repo and service classes.

Closes #6

denniscrevus committed:
feat: 🎸 departmentRepo + departmentService created

✅ Closes: #6

sebastianbadea197 committed:
feat: 🎸 Add the corporatist service and repo classes

✅ Closes: #6

denisionita committed:
feat: 🎸 Add Supervisor Class

✅ Closes: #5

denisionita committed:
feat: 🎸 Add SurpevisorRepo

sebastianbadea197 committed:
Merge branch 'development' into feat/#6-create-the-repo-interface-and-the-service-class

denniscrevus committed:
feat: 🎸 Updated departmentService. Added comments

✅ Closes: #6

sebastianbadea197 committed:
feat: 🎸 Add the service and repo classes

✅ Closes: #6

CristiPV commented on apps/java-api/src/main/java/org/scoalaonline/api/model/Department.java:
```diff
@@ -38,7 +38,7 @@ public long getId() {
    * @return the name of the department
    */
-  public String getDepartmentName() {
+  public String getName() {
```
```suggestion
  public String getDepartmentName() {
```

CristiPV commented on apps/java-api/src/main/java/org/scoalaonline/api/model/Department.java:
```diff
@@ -60,7 +60,7 @@ public void setId(long id) {
    * Setter for the name of the department
    */
-  public void setDepartmentName(String departmentName) {
+  public void setName(String departmentName) {
```
```suggestion
  public void setDepartmentName(String departmentName) {
```

CristiPV commented on apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java:
```diff
@@ -0,0 +1,105 @@
+    corporatistToAdd.setEmail(corporatist.getEmail());
+    corporatistToAdd.setDepartment(corporatist.getDepartment());
+    corporatistToAdd.setSupervisor(corporatist.getSupervisor());
+    corporatistToAdd.setId(corporatist.getId());
```
```suggestion
```

sebastianbadea197 commented:
🚀🚀🚀

CristiPV commented on apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java:
```diff
@@ -0,0 +1,105 @@
+      if (corporatist.getFirstName() != null) {
+        corporatistToUpdate.setFirstName(corporatist.getFirstName());
+      }
+        corporatistToUpdate.setMiddleName(corporatist.getMiddleName());
```
```suggestion
      if (corporatist.getMiddleName() != null) {
        corporatistToUpdate.setMiddleName(corporatist.getMiddleName());
      }
```

CristiPV commented on apps/java-api/src/main/java/org/scoalaonline/api/service/DepartmentService.java:
```diff
@@ -0,0 +1,77 @@
+  public Department addDepartment ( Department department ) {
+    Department departmentToAdd = new Department();
+    departmentToAdd.setName( department.getName() );
+    departmentToAdd.setId( department.getId() );
```
```suggestion
```

CristiPV commented on apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java:
```diff
@@ -0,0 +1,99 @@
+  public Supervisor addSupervisor (Supervisor supervisor) {
+    Supervisor supervisorToAdd = new Supervisor();
+
+    supervisorToAdd.setId(supervisor.getId());
```
```suggestion
```

CristiPV commented on apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java:
```diff
@@ -0,0 +1,99 @@
+      }
+      return supervisorRepo.save(supervisorToUpdate);
+    } catch (NoSuchElementException e) {
+      System.out.println("The given id doesn't link to any supervisor.");
```
```suggestion
      System.out.println("The supervisor to update doesn't exist.");
```

sebastianbadea197 committed:
Update apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java

Co-authored-by: CristiPV <cristi.valentin29@gmail.com>

sebastianbadea197 committed:
Update apps/java-api/src/main/java/org/scoalaonline/api/service/SupervisorService.java

Co-authored-by: CristiPV <cristi.valentin29@gmail.com>

sebastianbadea197 committed:
Update apps/java-api/src/main/java/org/scoalaonline/api/service/DepartmentService.java

Co-authored-by: CristiPV <cristi.valentin29@gmail.com>

sebastianbadea197 committed:
Update apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java

Co-authored-by: CristiPV <cristi.valentin29@gmail.com>

sebastianbadea197 committed:
Update apps/java-api/src/main/java/org/scoalaonline/api/service/CorporatistService.java

Co-authored-by: CristiPV <cristi.valentin29@gmail.com>

sebastianbadea197 committed:
fix: 🐛 typo fix for department name

changed getName and setName into getDepartmentName and setDepartmentName

✅ Closes: #6

Vimpat commented:
¯\_(ツ)_/¯

sebastianbadea197 commented:
¯\\_(ツ)_/¯
